src/Zone/index.d.ts

```diff
@@ -20,22 +20,49 @@ interface Zone {
 	// Methods
 	findLocalPlayer(): boolean;
 	findPlayer(player: Player): boolean;
-	findPart(part: BasePart): boolean;
+	findPart(part: BasePart): LuaTuple<[boolean, BasePart[]]>;
+	findItem(characterOrBasePart: BasePart | Model): LuaTuple<[boolean, BasePart[]]>;
+	findPoint(point: Vector3 | CFrame): LuaTuple<[boolean, BasePart[]]>;
 	getPlayers(): Player[];
 	getParts(): BasePart[];
+	getItems(): Array<BasePart | Model>;
 	/**
 	 * Generates random points within the zones region until one falls within its bounds.
 	 * It then returns this Vector3 and a table array of group parts the point falls within.
 	 */
-	getRandomPoint(): [Vector3, BasePart[]];
+	getRandomPoint(): LuaTuple<[Vector3, BasePart[]]>;
 	/**
-	 * Sets the frequency of checks based upon the Accuracy Enum.
+	 * This is used to detect your own custom instances within zones, such as NPCs, and is a recommended replacement for part-events/methods.
+	 * An item can be any BasePart or Character/NPC (i.e. a model with a Humanoid and HumanoidRootPart). Once tracked, it can be listened for with the zone.itemEntered and zone.itemExited events.
+	 * An item will be automatically untracked if destroyed or has its parent set to nil.
+	 * @param item
 	 */
-	setAccuracy(enumIdOrName: Accuracy[keyof Accuracy] | keyof Accuracy): void;
+	trackItem(characterOrBasePart: BasePart | Model): void;
+	untrackItem(characterOrBasePart: BasePart | Model): void;
 	/**
-	 * Sets the precision of checks based upon the [Detection Enum]. Defaults to 'Automatic'.
+	 * This is used to bind the zone to a settingsGroup to enhance the default behaviour of a collection of zones. The properties of a settingsGroup can be viewed at and customised using ZoneController.setGroup.
+	 * This method is particularly useful for zones where you want to guarantee the player/item is not in two zones at once. For example, when working with ambient/music/lighting zones which perfectly border each other.
+	 */
+	bindToGroup(settingsGroupName: string): void;
+	unbindFromGroup(): void;
+	/**
+	 * Sets the precision of checks based upon the [Detection Enum]. Defaults to `Automatic`.
 	 */
 	setDetection(enumIdOrName: Detection[keyof Detection] | keyof Detection): void;
+	/**
+	 * Moves the zone outside of workspace into a separate WorldModel within ReplicatedStorage or ServerStorage. This action is irreversible - once called it cannot be undone.
+	 */
+	relocate(): void;
+	/**
+	 * Tracks the item until it has entered the zone, then calls the given function. If the item is already within the zone, the given function is called right away.
+	 * ```lua
+	 * local item = character:FindFirstChild("HumanoidRootPart")
+	 * zone:onItemEnter(item, function()
+	 *     print("The item has entered the zone!"))
+	 * end)
+	 * ```
+	 */
+	onItemEnter(characterOrBasePart: Model | BasePart, callback: () => void): void;
```

Comment:
The types for onItemExit seems to be missing:
https://github.com/christopher-buss/rbxts-zone-plus/blob/bf86aa852266e57643e4e87d657042d6c2752b64/src/Zone/init.lua#L864-L866